src/rebar_utils.erl

```diff
@@ -717,7 +717,10 @@ update_code(Paths, Opts) ->
                                   code:add_patha(Path),
                                   case lists:member(soft_purge, Opts) of
                                       true  ->
-                                          [begin code:soft_purge(M), code:delete(M) end || M <- Modules];
+                                            %% only soft purge modules that were not
+                                            %% cover compiled, otherwise coverage is lost
+                                          [begin code:soft_purge(M), code:delete(M) end ||
+                                                M <- Modules, cover:is_compiled(M) =:= false];
```

Comment:
Sounds good to me there. Does `cover:is_compiled(M)` always work, even when cover is not started yet?